config/secretsmanager/config.go

```diff
@@ -13,5 +22,193 @@ func Configure(p *config.Provider) {
 		r.TerraformConfigurationInjector = func(_ map[string]any, params map[string]any) {
 			params["name_prefix"] = ""
 		}
+		r.TerraformCustomDiff = func(diff *terraform.InstanceDiff, state *terraform.InstanceState, config *terraform.ResourceConfig) (*terraform.InstanceDiff, error) {
+			// skip diff customization on create
+			if state == nil || state.Empty() {
+				return diff, nil
+			}
+			if config == nil {
+				return nil, errors.New("resource config cannot be nil")
```

Comment:
If not nil, what is the `config` during a destroy call?

Reply:
Though destroy does not make use of it, it is always populated with the current parameters at `spec.forProvider`
https://github.com/crossplane/upjet/blob/87fae5ea613cb8908e976c8f2a8c966e3416f96e/pkg/controller/external_nofork.go#L418
and passed to `TerraformCustomDiff` function
https://github.com/crossplane/upjet/blob/87fae5ea613cb8908e976c8f2a8c966e3416f96e/pkg/controller/external_nofork.go#L424
For destroys, the indicator is the `Destroy` field of instanceDiff object
https://github.com/crossplane/upjet/blob/87fae5ea613cb8908e976c8f2a8c966e3416f96e/pkg/controller/external_nofork.go#L689